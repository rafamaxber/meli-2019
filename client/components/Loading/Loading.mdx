---
name: Loading
menu: Components
---

import { Playground, Props } from 'docz'
import { MainThemeProvider } from '../Layouts/ThemeProvider.tsx'

import Loading from './Loading.tsx'

# Loading

## Basic usage

<Playground>
  <MainThemeProvider>
    <Loading />
  </MainThemeProvider>
</Playground>

