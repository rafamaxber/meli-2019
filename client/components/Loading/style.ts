import styled from 'styled-components';

export const Loading = styled.div`
  text-align: center;
  min-height: 600px;
  padding-top: 20px;
`;

export const LoadingIcon = styled.img`
  width: 60px;
`;
