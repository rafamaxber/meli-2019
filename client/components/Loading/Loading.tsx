import * as Common from '../Common/main.styled'
import * as Styled from './style'

const Loading = () => (
  <Common.Limit>
    <Styled.Loading>
      <Styled.LoadingIcon
        src="/static/img/spinner.gif"
        title="Carregando..."
        alt="Carregando..."
      />
    </Styled.Loading>
  </Common.Limit>
)

export default Loading
