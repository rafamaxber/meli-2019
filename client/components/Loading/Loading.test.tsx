import React from 'react';
import { render } from '@testing-library/react';
import Loading from './Loading';

describe('Loading', () => {
  describe('smoke tests', () => {
    test('should render component with Loading', () => {
      const component = render(<Loading/>);

      expect(component).toMatchSnapshot();
    })
  });
})


