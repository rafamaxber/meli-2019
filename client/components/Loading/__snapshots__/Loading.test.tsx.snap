// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading smoke tests should render component with Loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  text-align: center;
  min-height: 600px;
  padding-top: 20px;
}

.c1 {
  width: 60px;
}

<body>
    <div>
      <div
        class=""
      >
        <div
          class="c0"
        >
          <img
            alt="Carregando..."
            class="c1"
            src="/static/img/spinner.gif"
            title="Carregando..."
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mainstyled__Limit-kgem6r-2 poVdl"
    >
      <div
        class="style__Loading-sc-1sdhoa5-0 iwQOtz"
      >
        <img
          alt="Carregando..."
          class="style__LoadingIcon-sc-1sdhoa5-1 hzkFxQ"
          src="/static/img/spinner.gif"
          title="Carregando..."
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
